main.c

```diff
@@ -1,16 +1,17 @@
-
+#include <hal/debug.c>
```

Comment:
In addition, try to `#define printf(...) debugPrint(__VA_ARGS__)` for NXDK.
A good alternative if that doesn't work, is to remap stdout to a log.txt (see my OpenSWE1R Xbox port for that).